The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12870340172/job/35881157424) [(plain)](https://github.com/rust-lang/rust/commit/7fe9910b41f8ae1ebe61f4a6c7d6ee64f03258f9/checks/35881157424/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling compiler_builtins v0.1.143
error: unresolved link to `LendingIterator`
    --> library/core/src/slice/mod.rs:1104:19
     |
1104 |     /// though a [LendingIterator]-equivalent is possible.
     |                   ^^^^^^^^^^^^^^^ no item named `LendingIterator` in scope
     = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
     = note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`
     = help: to override `-D warnings` add `#[allow(rustdoc::broken_intra_doc_links)]`


error: unresolved link to `LendingIterator`
    --> library/core/src/slice/mod.rs:1107:10
     |
1107 |     /// [LendingIterator]: https://blog.rust-lang.org/2022/10/28/gats-stabilization.html
     |          ^^^^^^^^^^^^^^^ no item named `LendingIterator` in scope
     = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

error: this URL is not a hyperlink
    --> library/core/src/slice/mod.rs:1107:28
    --> library/core/src/slice/mod.rs:1107:28
     |
1107 |     /// [LendingIterator]: https://blog.rust-lang.org/2022/10/28/gats-stabilization.html
     |
     = note: bare URLs are not automatically turned into clickable links
     = note: `-D rustdoc::bare-urls` implied by `-D warnings`
     = help: to override `-D warnings` add `#[allow(rustdoc::bare_urls)]`
     = help: to override `-D warnings` add `#[allow(rustdoc::bare_urls)]`
help: use an automatic link instead
     |
1107 |     /// [LendingIterator]: <https://blog.rust-lang.org/2022/10/28/gats-stabilization.html>

error: could not document `core`
warning: build failed, waiting for other jobs to finish...
warning: build failed, waiting for other jobs to finish...
Command has failed. Rerun with -v to see more details.
  local time: Mon Jan 20 15:04:11 UTC 2025
  network time: Mon, 20 Jan 2025 15:04:11 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/slice/mod.rs
+++ b/library/core/src/slice/mod.rs
@@ -1099,10 +1099,15 @@ impl<T> [T] {
     /// assert!(iter.next().is_none());
     /// ```
     ///
-    /// There's no `windows_mut`, as that existing would let safe code violate the
-    /// "only one `&mut` at a time to the same thing" rule.  However, you can sometimes
-    /// use [`Cell::as_slice_of_cells`](crate::cell::Cell::as_slice_of_cells) in
-    /// conjunction with `windows` to accomplish something similar:
+    /// Because the [Iterator] trait cannot represent the required lifetimes,
+    /// there is no `windows_mut` analog to `windows`;
+    /// `[0,1,2].windows_mut(2).collect()` would violate [the rules of references]
+    /// (though a [LendingIterator] analog is possible). You can sometimes use
+    /// [`Cell::as_slice_of_cells`](crate::cell::Cell::as_slice_of_cells) in
+    /// conjunction with `windows` instead:
+    ///
+    /// [the rules of references]: https://doc.rust-lang.org/book/ch04-02-references-and-borrowing.html#the-rules-of-references
+    /// [LendingIterator]: https://blog.rust-lang.org/2022/10/28/gats-stabilization.html
     /// ```
     /// use std::cell::Cell;
     ///

